File: ClassroomServer-Magazine/Models/Instructor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

public enum AuthorityRank
{
    Low,
    Medium,
    High
}

namespace ClassroomServer.Models
{
    public class Instructor
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public AuthorityRank AuthRank { get; set; } = AuthorityRank.Low;

        [Required]
        public string FirstName { get; set; } = null!;

        [Required]
        public string LastName { get; set; } = null!;

        public List<Guid> CourseIds { get; set; } = new();
        public virtual ICollection<Course> Courses { get; set; } = new List<Course>();
    }
}

File: ClassroomMgmt-Magazine/src/models/Instructor.ts
import { Course, NEW_COURSE_TEMPLATE } from "./Course";

export interface Instructor {
  id?: string;
  firstName: string;
  lastName: string;
  courseIds: string[];
}

export const EMPTY_INSTRUCTOR: Instructor = {
  id: "0",
  firstName: "",
  lastName: "",
  courseIds: [""],
};
